### DISCUSSION TITLE
elastalert_error index mapping

### DISCUSSION THREAD

**Discussion by vvvlad** [jertel/elastalert2#274] [Q&A]
Good day everyone.
Looking for some understanding of elastalert_error index mapping.
The index keeps the failed rule name under data.name.
The mapping is:
```
"data" : {
  "type" : "object",
  "enabled" : false
}
```
which AFAIK means that there is no way to search by failed index name.
What is the reason for this mapping and is it possible to change it in the elastalert configuration?

Thanks

**Comment by nsano-rururu** [comment]
Did you read the source code?
https://github.com/jertel/elastalert2/blob/master/elastalert/create_index.py

**Reply by nsano-rururu** [reply]
Have you never used elastalert2? .. Or is it there?

**Reply by vvvlad** [reply]
Here is the definition:
https://github.com/jertel/elastalert2/blob/master/elastalert/es_mappings/6/elastalert_error.json

The question is why? And how I change it if I just want to pip install the package without changing internal files.

**Reply by nsano-rururu** [reply]
Don't you feel like talking? all right. Goodbye.

**Reply by vvvlad** [reply]
@nsano-rururu
I don't understand what you mean.

**Reply by nsano-rururu** [reply]
I don't want to get involved

**Reply by vvvlad** [reply]
> I don't want to get involved

Ok, I'm not sure what you mean by getting involved. 

Thanks for pointing me to the source, that helped me to find the definition file.

**Reply by nsano-rururu** [reply]
I haven't seen the source code, and I understand that I'm not technically familiar with this point.

**Reply by nsano-rururu** [reply]
>how I change it if I just want to pip install the package without changing internal files.

See the following documentation for pip installation.
https://elastalert2.readthedocs.io/en/latest/running_elastalert.html#as-a-python-package
